core/src/main/java/com/linecorp/armeria/server/DefaultServiceBuilder.java

```diff
@@ -0,0 +1,170 @@
+        return this;
+    }
+
+    @Override
+    public ServiceBuilder verboseResponses(boolean verboseResponses) {
+        this.verboseResponses = verboseResponses;
+        return this;
+    }
+
+    @Override
+    public ServiceBuilder requestContentPreviewerFactory(ContentPreviewerFactory factory) {
+        requestContentPreviewerFactory = requireNonNull(factory, "factory");
+        return this;
+    }
+
+    @Override
+    public ServiceBuilder responseContentPreviewerFactory(ContentPreviewerFactory factory) {
+        responseContentPreviewerFactory = requireNonNull(factory, "factory");
+        return this;
+    }
+
+    @Override
+    public ServiceBuilder contentPreview(int length) {
+        return contentPreview(length, ArmeriaHttpUtil.HTTP_DEFAULT_CONTENT_CHARSET);
+    }
+
+    @Override
+    public ServiceBuilder contentPreview(int length, Charset defaultCharset) {
+        return contentPreviewerFactory(ContentPreviewerFactory.ofText(length, defaultCharset));
+    }
+
+    @Override
+    public ServiceBuilder contentPreviewerFactory(ContentPreviewerFactory factory) {
+        requestContentPreviewerFactory(factory);
+        responseContentPreviewerFactory(factory);
+        return this;
+    }
+
+    @Override
+    public ServiceBuilder accessLogFormat(String accessLogFormat) {
+        return accessLogWriter(AccessLogWriter.custom(requireNonNull(accessLogFormat, "accessLogFormat")),
+                               true);
+    }
+
+    @Override
+    public ServiceBuilder accessLogWriter(AccessLogWriter accessLogWriter,
+                                          boolean shutdownOnStop) {
+        this.accessLogWriter = requireNonNull(accessLogWriter, "accessLogWriter");
+        shutdownAccessLogWriterOnStop = shutdownOnStop;
+        return this;
+    }
+
+    @Override
+    public <T extends Service<HttpRequest, HttpResponse>, R extends Service<HttpRequest, HttpResponse>>
+    ServiceBuilder decorator(Function<T, R> decorator) {
+        requireNonNull(decorator, "decorator");
+
+        @SuppressWarnings("unchecked")
+        final Function<Service<HttpRequest, HttpResponse>, Service<HttpRequest, HttpResponse>> castDecorator =
+                (Function<Service<HttpRequest, HttpResponse>, Service<HttpRequest, HttpResponse>>) decorator;
+
+        if (this.decorator != null) {
+            this.decorator = this.decorator.andThen(castDecorator);
+        } else {
+            this.decorator = castDecorator;
+        }
+
+        return this;
+    }
+
+    public Service<HttpRequest, HttpResponse> decorate(Service<HttpRequest, HttpResponse> service) {
+        if (decorator == null) {
+            return service;
+        }
+        return service.decorate(decorator);
+    }
+
+    public ServiceConfigBuilder serviceConfigBuilder(Route route, Service<HttpRequest, HttpResponse> service) {
+        final ServiceConfigBuilder serviceConfigBuilder = new ServiceConfigBuilder(route, service);
```

Comment:
I could decorate a service with the decorate here, but `AnnotatedServiceBindingBuilder` needs exceptionHandlingDecorators as the last decorator https://github.com/line/armeria/pull/2180/files#diff-162326dc40b1d719dcad478d6ac513bcR185

Reply:
I see. How about adding that as a Javadoc of this method?